spyder/api/shellconnect/main_widget.py

```diff
@@ -17,13 +24,34 @@
 from spyder.api.widgets.main_widget import PluginMainWidget
 from spyder.widgets.emptymessage import EmptyMessageWidget
 
+if TYPE_CHECKING:
+    from qtpy.QtWidgets import QWidget
+
+    import spyder.plugins.ipyconsole.widgets.shell
+
 
 class _ErroredMessageWidget(EmptyMessageWidget):
     """Widget to show when the kernel's shell failed to start."""
 
-    def __init__(self, parent, shellwidget):
-        # Initialize EmptyMessageWidget with the content we want to show for
-        # errors
+    def __init__(
+        self,
+        parent: ShellConnectMainWidget,
+        shellwidget: spyder.plugins.ipyconsole.widgets.shell.ShellWidget,
+    ) -> None:
+        """
+        Initialize :class:`EmptyMessageWidget` with content to show for errors.
+
+        Parameters
+        ----------
+        parent : ShellConnectMainWidget
+            The parent widget of this one.
+        shellwidget : spyder.plugins.ipyconsole.widgets.shell.ShellWidget
+            The kernel shell widget that failed to launch.
```

Comment:
```suggestion
            The shell widget that failed to launch.
```

Makes no sense to mention `kernel` when talking about widgets (the kernel is a separate Python process to which the shell widget is connected).